src/mbgl/renderer/render_orchestrator.cpp

```diff
@@ -361,26 +361,25 @@ std::unique_ptr<RenderTree> RenderOrchestrator::createRenderTree(const UpdatePar
         }
 
         bool symbolBucketsChanged = false;
-        renderTreeParameters->placementChanged = !placement->stillRecent(updateParameters.timePoint, updateParameters.transformState.getZoom());
+        for (auto it = layersNeedPlacement.rbegin(); it != layersNeedPlacement.rend(); ++it) {
+            auto result = crossTileSymbolIndex.addLayer(*it, updateParameters.transformState.getLatLng().longitude()); 
+            symbolBucketsChanged = symbolBucketsChanged || (result & CrossTileSymbolIndex::AddLayerResult::BucketsAdded);
```

Comment:
Should we make new placement when buckets were removed `BucketsRemoved`? So that, for example, when obscuring topmost symbol layer is hidden, symbols from underlying layers could be immediately placed?

Reply:
Thanks for the catch! I think we should indeed update opacities on any change (but the new placement is still to be initiated only on new buckets added).